app/src/main/java/com/sun/heartrate/data/source/asynctask/HeartTask.kt

```diff
@@ -1,26 +1,25 @@
 package com.sun.heartrate.data.source.asynctask
 
 import android.os.AsyncTask
-import com.sun.heartrate.data.model.HeartModel
 import com.sun.heartrate.data.model.OnDataLoadedCallback
 
-abstract class HeartTask(
-    private val callback: OnDataLoadedCallback<List<HeartModel>>
-) : AsyncTask<Any?, Any?, List<HeartModel>>() {
+abstract class HeartTask<T>(
+    private val callback: OnDataLoadedCallback<T>
+) : AsyncTask<Any?, Any?, T>() {
     
     private var exception: Exception? = null
     
-    abstract fun getData(): List<HeartModel>
+    abstract fun getData(): T
```

Comment:
Em thấy getData() chưa hợp lý lắm =)) 
```suggestion
    abstract fun onExecute(): T
```